IdentityCore/src/throttling/cache/MSIDThrottlingLRUCache.h

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+@class MSIDThrottlingCacheRecord;
+
+@protocol MSIDThrottlingLRUCache <NSObject>
+
+@property (nonatomic, readonly) NSUInteger cacheSize;
+
+- (instancetype)initWithCacheSize:(NSUInteger)cacheSize;
+
+//NOTE: We don't need to pass in throttle type here, because we already know it will be 429/5xx if it goes into serverDelayCache
+//and UI_Required if it goes into UIRequiredCache, so we pass in those info into CacheNode objects within each subclass' virtual methods.
```

Comment:
What if we need to add new type of cache in the future? For example, if we want to throttle invalid_request errors. Can we make the design more extensible for new error types?